src/Features/Traits/Eloquent.php

```diff
@@ -4,23 +4,146 @@
 
 namespace Blumilk\BLT\Features\Traits;
 
+use Behat\Gherkin\Node\TableNode;
+use Blumilk\BLT\Helpers\LaravelRelations;
 use Illuminate\Support\Str;
 use PHPUnit\Framework\Assert;
 
 trait Eloquent
 {
     /**
-     * @Given there is :count :model object in database
-     * @Given there should be :count :model object in database
-     * @Given there are :count :model objects in database
-     * @Given there should be :count :model objects in database
+     * @Given there is a model :model in the database
+     * @Given there is a :model in the database
      */
-    public function thereAreModelsInDatabase(string $model, int $count): void
+    public function seedModelInTheDatabase(string $model, ?TableNode $table = null): void
     {
         $modelClass = $this->recognizeModelClass($model);
+        $attributes = $table ? $table->getRowsHash() : [];
+
+        if (method_exists($modelClass, "factory")) {
+            $modelClass::factory()->create($attributes);
+        } else {
+            $modelClass::create($attributes);
+        }
+    }
+
+    /**
+     * @Then the model :model exists in the database
+     */
+    public function assertModelExistsInTheDatabase(string $model): void
+    {
+        $modelClass = $this->recognizeModelClass($model);
+
+        Assert::assertTrue($modelClass::query()->exists(), "The model $model does not exist in the database.");
+    }
+
+    /**
+     * @Given there are :count :model objects in the database
+     * @Given there are :count :model in the database
+     */
+    public function thereAreModelsInTheDatabase(string $model, int $count): void
+    {
+        $modelClass = $this->recognizeModelClass($model);
+        $existingCount = $modelClass::query()->count();
+
+        if ($existingCount < $count) {
+            $modelClass::factory()->count($count - $existingCount)->create();
+        }
+
         Assert::assertEquals($count, $modelClass::query()->count());
     }
 
+    /**
+     * @Then the model :model1 has many :model2
+     */
+    public function theModelHasMany(string $model1, string $model2): void
+    {
+        $model1Class = $this->recognizeModelClass($model1);
+        $relation = Str::plural($model2);
+        $instance = $model1Class::first() ?: $model1Class::factory()->create();
+
+        Assert::assertTrue(
+            method_exists($instance, $relation),
+            "The model $model1 does not have a $relation relation method.",
+        );
+
+        LaravelRelations::assertRelation($instance, $relation, 'Illuminate\Database\Eloquent\Relations\HasMany');
```

Comment:
Please look at this file: https://github.com/blumilksoftware/blt/blob/main/src/LaravelContracts.php and try to replicate its purpose here.